internal/app/components/roles/roles.go

```diff
@@ -24,3 +26,7 @@ func GetRoleFromId(id RoleId) Role {
 	}
 	return Role(NewConrad())
 }
+
+func GetRoleFromContext(ctx context.Context) Role {
+	return ctx.Value(contextKeys.UserRoleKey).(Role)
```

Comment:
If the route is authenticated permission level or above then yes, because the middleware is supposed to put role in each context. However, we should check for the nil case just to be safe, good catch.